capeinfra/swimlanes/private.py

```diff
@@ -407,3 +417,190 @@ def create_dap_submission_queue(self):
+
+            self.vpn_server_acm_cert = aws.acm.Certificate(
+                f"{self.basename}-vpn-srvracmcert",
+                certificate_chain=ca_crt_pem,
+                private_key=server_key_pem,
+                certificate_body=server_cert_pem,
+                opts=ResourceOptions(parent=self),
+            )
+        except FileNotFoundError as fnfe:
+            raise ValueError(
+                f"One or more configured TLS asset files could not be found "
+                f"during VPN configuration: {fnfe}"
+            )
+
+    # TODO: ISSUE #100
+    def create_vpn(self):
+        """Creates/configures a Client VPN Endpoint for the private swimlane.
+
+        In the case that the required configuration for the VPN ACM certificate
+        does not exist or has issues, a message will be printed to the pulumi
+        console and VPN configuration will cease.
+
+        """
+
+        try:
+            self.create_vpn_acm_certificate()
+        except ValueError as ve:
+            warn(
+                f"Error encountered in configuration of Private swimlane VPN. "
+                f"{ve}. VPN will not be configured."
+            )
+            return
+
+        # log group and log stream for the VPN endpoint
+        self.vpn_log_group = aws.cloudwatch.LogGroup(
+            f"{self.basename}-vpn-logs",
+            name=f"{self.basename}-vpn-logs",
+            tags={"desc_name": (f"{self.desc_name} VPN Log Group")},
+            opts=ResourceOptions(parent=self),
+        )
+
+        self.vpn_log_stream = aws.cloudwatch.LogStream(
+            f"{self.basename}-vpn-logs-stream",
+            name=f"{self.basename}-vpn-logs-stream",
+            log_group_name=self.vpn_log_group.name,
+            opts=ResourceOptions(parent=self),
+        )
+
+        # Client VPN endpoint, which is the interface VPN connections go thu
+        self.client_vpn_endpoint = aws.ec2clientvpn.Endpoint(
+            f"{self.basename}-vpnep",
+            description=f"{self.desc_name} Client VPN Endpoint",
+            server_certificate_arn=self.vpn_server_acm_cert.arn,
+            authentication_options=[
+                {
+                    "type": "certificate-authentication",
+                    "root_certificate_chain_arn": self.vpn_server_acm_cert.arn,
+                }
+            ],
+            client_cidr_block=self.config.get(
+                "vpn", "cidr-block", default="10.1.0.0/22"
+            ),
+            connection_log_options={
+                "enabled": True,
+                "cloudwatch_log_group": self.vpn_log_group.name,
+                "cloudwatch_log_stream": self.vpn_log_stream.name,
+            },
+            tags={
+                "desc_name": (
+                    f"{self.desc_name} DAP submission sqs message lambda trigger "
+                    "function"
+                )
+            },
+            transport_protocol=self.config.get(
+                "vpn", "transport-proto", default="udp"
```

Comment:
I think this default should really be captured in the default configuration rather than hard coding it here